modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/AssetHandler.java

```diff
@@ -0,0 +1,106 @@
+package com.enonic.xp.portal.impl.handler;
+
+import java.util.EnumSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+
+import com.google.common.net.HttpHeaders;
+
+import com.enonic.xp.app.ApplicationKey;
+import com.enonic.xp.core.internal.HexCoder;
+import com.enonic.xp.portal.PortalResponse;
+import com.enonic.xp.portal.impl.PortalConfig;
+import com.enonic.xp.resource.Resource;
+import com.enonic.xp.resource.ResourceKey;
+import com.enonic.xp.resource.ResourceService;
+import com.enonic.xp.server.RunMode;
+import com.enonic.xp.util.MediaTypes;
+import com.enonic.xp.web.HttpMethod;
+import com.enonic.xp.web.WebException;
+import com.enonic.xp.web.WebRequest;
+import com.enonic.xp.web.WebResponse;
+
+import static com.google.common.base.Strings.nullToEmpty;
+
+@Component(service = AssetHandler.class, configurationPid = "com.enonic.xp.portal")
+public class AssetHandler
+{
+    private static final Pattern PATTERN = Pattern.compile( "([^/^:]+)(?::([^/]+))?/(.+)" );
```

Comment:
this pattern looks incorrect. Does not have a start, and disallows `^` in first path element?